compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -2666,8 +2664,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     }
 
     /// Returns the parameters of a function, with their generic parameters if those are the full
-    /// type of that parameter. Returns `None` if the function has no generics or the body is
-    /// unavailable (eg is an instrinsic).
+    /// type of that parameter.
+    ///
+    /// Returns `None` if the function has no parameters taking generic type, or the function is
```

Comment:
This doesn't really make sense...? I think it's meaningful to return spans for functions if they don't have generics... like, people can still make argument mismatch errors if the function is not generic 🤔

Reply:
This is specifically for, and only used for, generic mismatches. It doesn't really make sense to gather the generic parameters outside that context.

Reply:
We still definitely want to label mismatched parameters even if those parameters are not associated with generic parameters, though.

Reply:
Reverted as part of the other issue.